src/linker/Linker.Steps/ResolveFromXmlStep.cs

```diff
@@ -47,15 +47,18 @@ public class ResolveFromXmlStep : ProcessLinkerXmlStepBase
 
 		static readonly string[] _accessorsAll = new string[] { "all" };
 		static readonly char[] _accessorsSep = new char[] { ';' };
+		IMarker Marker { get; }
```

Comment:
I still don't understand why this abstraction is needed. It should be fine to keep the code as-is and it'd work as annotations are always populated before assembly marking

Reply:
It's so that `ResolveFromXmlStep` can run on XML descriptors from delay loaded assemblies. In this case it needs to fully mark XML dependencies (IMarker will be the MarkStep itself), instead of just marking annotations (the Initialize logic won't run again, nor should it).

Reply:
This has been removed